controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -0,0 +1,131 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "dynamic_channel_selection_task.h"
+
+#include "../son_actions.h"
+
+#include <bcl/beerocks_defines.h>
+#include <bcl/network/network_utils.h>
+#include <easylogging++.h>
+
+using namespace beerocks;
+using namespace net;
+using namespace son;
+
+#define FSM_MOVE_STATE(new_state)                                                                  \
+    ({                                                                                             \
+        TASK_LOG(TRACE) << "FSM: " << s_ar_states[int(m_fsm_state)] << " --> "                     \
+                        << s_ar_states[int(eState::new_state)];                                    \
+        m_fsm_state = eState::new_state;                                                           \
+    })
+
+#define FSM_IS_IN_STATE(state) (m_fsm_state == eState::state)
+#define FSM_CURR_STATE_STR s_ar_states[int(m_fsm_state)]
+
+#define EVENT_STR(event) s_ar_events[int(event)]
+
+const char *dynamic_channel_selection_task::s_ar_states[] = {FOREACH_DCS_STATE(GENERATE_STRING)};
+
+const char *dynamic_channel_selection_task::s_ar_events[] = {FOREACH_DCS_EVENT(GENERATE_STRING)};
+
+dynamic_channel_selection_task::dynamic_channel_selection_task(db &database_,
+                                                               ieee1905_1::CmduMessageTx &cmdu_tx_,
+                                                               task_pool &tasks_,
+                                                               std::string radio_mac_)
+    : database(database_), cmdu_tx(cmdu_tx_), tasks(tasks_), m_radio_mac(radio_mac_)
+{
+    m_fsm_state = eState::INIT;
```

Comment:
Why not move the `m_fsm_state` to the initializer list like all other members?

Reply:
not sure we need to start the task in a specific state.
I think we can remove it or just initialize m_fsm_state  in the header to INIT.

Reply:
nipick anyway